### PR TITLE
Mes 4709 activity code 4 modal

### PR DESCRIPTION
**Pull request by Cullima**
## Description
- Creates Mod 1 modal for ending test with activity code 4  for either speed requirements or emergency stop (implementation will be a separate PR)

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
Screenshot 2020-02-03 at 12 02 38

Screenshot 2020-02-03 at 12 05 44

### CODE REVIEW

File: src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@IonicPage()
+@Component({
+  selector: 'activity-code-4-modal',
+  templateUrl: 'activity-code-4-modal.html',
+})
+export class ActivityCode4Modal implements OnInit {
+
+  modalReason: ModalReason;
+
+  faultType: any;
```
Comment (matthew2564) [inline src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.ts]: is faultType used?

File: src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.constants.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export enum ModalReason {
+  SPEED_REQUIREMENTS = 'Speed requirements not met',
+  EMERGENCY_STOP = 'Serious fault recorded on emergency stop',
```
Comment (szabikr) [inline src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.constants.ts]: There is going to be another reason here:

1.  SPEED_REQUIREMENTS = 'Speed requirements not met'
2. EMERGENCY_STOP_SERIOUS = 'Serious fault recorded on emergency stop'
3. EMERGENCY_STOP_DANGEROUS = 'Dangerous fault recorded on emergency stop'

File: src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.ts (hunk 1 above)
Comment (Cullima) [reply src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.ts]: good spot. removed, thanks

File: src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.constants.ts (hunk 2 above)
Comment (Cullima) [reply src/pages/test-report/cat-a-mod1/components/activity-code-4-modal/activity-code-4-modal.constants.ts]: Added in

**Review by szabikr — APPROVED** [review]
Great stuff!

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**